algorithms/estonian/stemmer.tt

```diff
@@ -0,0 +1,479 @@
+		<DD>delete
+    <DT><B><I>mata</I></B>
+		<DD>delete
+    <DT><B><I>takse  &nbsp;  dakse</I></B>
+		<DD>delete
+    <DT><B><I>taks  &nbsp;  daks</I></B>
+		<DD>delete
+    <DT><B><I>akse</I></B>
+		<DD>delete
+    <DT><B><I>sime</I></B>
+		<DD>delete
+    <DT><B><I>site</I></B>
+		<DD>delete
+    <DT><B><I>sin</I></B>
+		<DD>delete
+    <DT><B><I>me preceded by </I>V1</B>
+		<DD>delete
+    <DT><B><I>da preceded by </I>V1</B>
+		<DD>delete
+    <DT><B><I>n preceded by </I>V1</B>
+		<DD>delete
+    <DT><B><I>b preceded by </I>V1</B>
+		<DD>delete
+		
+		<DD>set is_verb
+</DL>
+</DL>
+
+<p>
+Step 3: special_noun_endings
+</p>
+
+<DL><DD>
+    Search for the longest among the following suffixes in <I>R1</I>, and perform the
+    action indicated
+<BR><BR>
+<DL>
+    <DT><B><I>lasse</I></B>
+		<DD>replace by <B><I>lase</I></B>
+    <DT><B><I>last</I></B>
+		<DD>replace by <B><I>lase</I></B>
+    <DT><B><I>lane</I></B>
+		<DD>replace by <B><I>lase</I></B>
+    <DT><B><I>lasi</I></B>
+		<DD>replace by <B><I>lase</I></B>
+    <DT><B><I>misse</I></B>
+		<DD>replace by <B><I>mise</I></B>
+    <DT><B><I>mist</I></B>
+		<DD>replace by <B><I>mise</I></B>
+    <DT><B><I>mine</I></B>
+		<DD>replace by <B><I>mise</I></B>
+    <DT><B><I>misi</I></B>
+		<DD>replace by <B><I>mise</I></B>
+    <DT><B><I>lisse</I></B>
+		<DD>replace by <B><I>lise</I></B>
+    <DT><B><I>list</I></B>
+		<DD>replace by <B><I>lise</I></B>
+    <DT><B><I>line</I></B>
+		<DD>replace by <B><I>lise</I></B>
+    <DT><B><I>lisi</I></B>
+		<DD>replace by <B><I>lise</I></B>
+</DL>
+</DL>
+
+<p>
+Step 4: case_ending
+</p>
+
+<DL><DD>
+    Search for the longest among the following suffixes in <I>R1</I>, and perform the
+    action indicated
+<BR><BR>
+<DL>
+    <DT><B><I>sse</I> preceded by <I>RV</I> or <I>LONGV</I></B>
+    <DT><B><I>st</I> preceded by <I>RV</I> or <I>LONGV</I></B>
+    <DT><B><I>le</I> preceded by <I>RV</I> or <I>LONGV</I></B>
+    <DT><B><I>lt</I> preceded by <I>RV</I> or <I>LONGV</I></B>
+    <DT><B><I>ga</I> preceded by <I>RV</I> or <I>LONGV</I></B>
+    <DT><B><I>ks</I> preceded by <I>RV</I> or <I>LONGV</I></B>
+    <DT><B><I>ta</I> preceded by <I>RV</I> or <I>LONGV</I></B>
+    <DT><B><I>t</I></B>
+    <DT><B><I>s</I> preceded by <I>RV</I> or <I>LONGV</I></B>
+    <DT><B><I>l</I> preceded by <I>RV</I> or <I>LONGV</I></B>
+		<DD>delete</B>
+</DL>
+</DL>
+
+<p>
+Step 5: plural_three_first_cases
+</p>
+
+<DL><DD>
+    Search for the longest among the following suffixes in <I>R1</I>, and perform the
+    action indicated
+<BR><BR>
+<DL>
+    <DT><B><I>ikkude</I></B>
+		<DD>replace by <B><I>iku</I></B>
+    <DT><B><I>ikke</I></B>
+		<DD>replace by <B><I>iku</I></B>
+    <DT><B><I>ike</I></B>
+		<DD>replace by <B><I>iku</I></B>
+    <DT><B><I>sid</I></B>
+		<DD>if it is not preceded by <I>LONGV</I>, delete
+    <DT><B><I>te</I></B>
+		<DD>if it doesn't have at least 4 letters before it, replace by <B><I>t</I></B>.
```

Comment:
If I read the code right, it actually seems to do this only if not preceded by `t`.  Which is right, the code or this description?  Or have I misread the code?